Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryILLPreprocess.py

```diff
@@ -397,67 +346,110 @@ def _foregroundWidths(self):
+            linePosition = mtd[peakWSName].readY(0)[0]
+            self._cleanup.cleanup(peakWSName)
+        return ws, linePosition
+
+    def _addSampleLogInfo(self, ws, linePosition):
+        """Add foreground indices and linePosition to the sample logs, names start with reduction."""
+        # Add the fractional workspace index of the beam position to the sample logs of ws.
+        run = ws.run()
+        run.addProperty(common.SampleLogs.LINE_POSITION, float(linePosition), True)
+        # Add foreground start and end workspace indices to the sample logs of ws.
+        hws = self._foregroundWidths()
+        beamPosIndex = int(numpy.rint(linePosition))
+        if beamPosIndex > 255:
+            beamPosIndex = 255
+            self.self.log().warning('Is it a monitor spectrum?')
+        sign = self._workspaceIndexDirection(ws)
+        startIndex = beamPosIndex - sign * hws[0]
+        endIndex = beamPosIndex + sign * hws[1]
+        if startIndex > endIndex:
+            endIndex, startIndex = startIndex, endIndex
+        run.addProperty(common.SampleLogs.FOREGROUND_START, int(startIndex), True)
+        run.addProperty(common.SampleLogs.FOREGROUND_CENTRE, int(beamPosIndex), True)
+        run.addProperty(common.SampleLogs.FOREGROUND_END, int(endIndex), True)
+        # Add two theta to the sample logs of ws.
+        if not self.getProperty(Prop.TWO_THETA).isDefault:
+            twoTheta = self.getProperty(Prop.TWO_THETA).value
+        else:
+            spectrum_info = ws.spectrumInfo()
+            twoTheta = numpy.rad2deg(spectrum_info.twoTheta(int(numpy.rint(linePosition))))
+        run.addProperty(common.SampleLogs.TWO_THETA, float(twoTheta), 'degree', True)
+        return twoTheta
+
+    def _moveDetector(self, ws, twoTheta, linePosition):
+        """Perform detector position correction for direct and reflected beams."""
+        if self.getProperty(Prop.DIRECT_LINE_WORKSPACE).isDefault:
+            return ws
+        # Reflected beam calibration
+        detectorMovedWSName = self._names.withSuffix('detectors_moved')
+        directLineWS = self.getProperty(Prop.DIRECT_LINE_WORKSPACE).value
+        directLinePosition = directLineWS.run().getLogData(common.SampleLogs.LINE_POSITION).value
+        args = {'InputWorkspace': ws,
+                'OutputWorkspace': detectorMovedWSName,
+                'EnableLogging': self._subalgLogging,
+                'DetectorComponentName': 'detector',
+                'LinePosition': linePosition,
+                'DirectLinePosition': directLinePosition,
+                'DirectLineWorkspace': directLineWS}
+        if self._instrumentName is not 'D17':
+            args['DetectorCorrectionType'] = 'RotateAroundSample'
+            args['DetectorFacesSample'] = True
+            args['PixelSize'] = common.pixelSize(self._instrumentName)
+        SpecularReflectionPositionCorrect(**args)
+        detectorMovedWS = mtd[detectorMovedWSName]
```

Comment:
There is no difference between D17 and Figaro here. We should drop the `dict` stuff and just call 
```python
detectorMovedWS = SpecularReflectionPositionCorrect(
    InputWorkspace=ws,
    OutputWorkspace=detectorMovedWSName,
    DetectorCompnentName='detector',
    DetectorCorrectionType='RotateAroundSample',
    DetectorFacesSample=True,
    PixelSize=common.pixelSize(self._instrumentName),
    LinePosition=linePosition,
    DirectLinePosition=directLinePosition,
    DirectLineWorkspace=directLineWS,
    EnableLogging=self._enableLogging)
```